TipCatDotNet.Api/Controllers/TransactionController.cs

```diff
@@ -0,0 +1,66 @@
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using CSharpFunctionalExtensions;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using TipCatDotNet.Api.Filters.Pagination;
+using TipCatDotNet.Api.Models.Payments;
+using TipCatDotNet.Api.Services.HospitalityFacilities;
+using TipCatDotNet.Api.Services.Payments;
+
+namespace TipCatDotNet.Api.Controllers
+{
+    [Authorize]
+    [Route("api/transactions")]
+    [Produces("application/json")]
+    public class TransactionController : BaseController
+    {
+        public TransactionController(IMemberContextService memberContextService, ITransactionService transactionService)
+        {
+            _transactionService = transactionService;
+            _memberContextService = memberContextService;
+        }
+
+
+        /// <summary>
+        /// Gets transactions by member.
+        /// </summary>
+        /// <returns></returns>
+        [HttpGet("last")]
+        [ProducesResponseType(typeof(List<TransactionResponse>), StatusCodes.Status200OK)]
+        [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status400BadRequest)]
+        public async Task<IActionResult> GetLast()
+        {
+            var (_, isFailure, memberContext, error) = await _memberContextService.Get();
+            if (isFailure)
+                return BadRequest(error);
+
+            return OkOrBadRequest(await _transactionService.Get(memberContext, new PaginationFilter(lastPageDesc, countLastTransactions)));
+        }
+
+
+        /// <summary>
+        /// Gets transactions pagination by member.
+        /// </summary>
+        /// <param name="filter">Pagination filter</param>
+        /// <returns></returns>
+        [HttpGet]
+        [ProducesResponseType(typeof(List<TransactionResponse>), StatusCodes.Status200OK)]
+        [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status400BadRequest)]
+        public async Task<IActionResult> Get([FromQuery] PaginationFilter filter)
```

Comment:
To use `skip` and `top` with an upper limit is a conventional solution these days, no need to overengineer

Reply:
You offer to remove pagination? Or you mean that there is more preferred solution for pagination?